python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -1586,3 +1586,25 @@ def test_parquet_dataset_partitions_piece_path_with_fsspec(tempdir):
     # ensure the piece path is also posix-style
     expected = path + "/data.parquet"
     assert dataset.pieces[0].path == expected
+
+
+def test_read_table_with_fspath(tempdir):
```

Comment:
I would maybe move this test to `tests/parquet/test_basic.py`, where there is a similar `test_multiple_path_types`, which also tests the handling of path-like objects. So you can put it next to that test.

Reply:
Ah, sorry about that. Thanks for correcting it.